zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java

```diff
@@ -0,0 +1,127 @@
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.exporter.tasks.incident;
+
+import co.elastic.clients.elasticsearch.ElasticsearchAsyncClient;
+import co.elastic.clients.elasticsearch._types.SortOrder;
+import co.elastic.clients.elasticsearch._types.query_dsl.Query;
+import co.elastic.clients.elasticsearch._types.query_dsl.QueryBuilders;
+import co.elastic.clients.elasticsearch.core.SearchRequest;
+import co.elastic.clients.elasticsearch.core.SearchResponse;
+import co.elastic.clients.elasticsearch.core.search.SourceFilter;
+import co.elastic.clients.json.JsonData;
+import io.camunda.exporter.tasks.incident.IncidentUpdateRepository.NoopIncidentUpdateRepository;
+import io.camunda.webapps.schema.descriptors.operate.template.PostImporterQueueTemplate;
+import io.camunda.webapps.schema.entities.operate.IncidentState;
+import io.camunda.webapps.schema.entities.operate.post.PostImporterActionType;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+
+public final class ElasticsearchIncidentUpdateRepository extends NoopIncidentUpdateRepository {
+
+  private final int partitionId;
+  private final String pendingUpdateAlias;
+  private final String incidentAlias;
+  private final String listViewAlias;
+  private final String flowNodeAlias;
+  private final ElasticsearchAsyncClient client;
+  private final Executor executor;
+
+  public ElasticsearchIncidentUpdateRepository(
+      final int partitionId,
+      final String pendingUpdateAlias,
+      final String incidentAlias,
+      final String listViewAlias,
+      final String flowNodeAlias,
+      final ElasticsearchAsyncClient client,
+      final Executor executor) {
+    this.partitionId = partitionId;
+    this.pendingUpdateAlias = pendingUpdateAlias;
+    this.incidentAlias = incidentAlias;
+    this.listViewAlias = listViewAlias;
+    this.flowNodeAlias = flowNodeAlias;
+    this.client = client;
+    this.executor = executor;
+  }
+
+  @Override
+  public CompletionStage<PendingIncidentUpdateBatch> getPendingIncidentsBatch(
+      final long fromPosition, final int size) {
+    final var query = createPendingIncidentsBatchQuery(fromPosition);
+    final var request = createPendingIncidentsBatchRequest(size, query);
+
+    return client
+        .search(request, PendingIncidentUpdate.class)
+        .thenApplyAsync(this::createPendingIncidentBatch, executor);
+  }
+
+  private SearchRequest createPendingIncidentsBatchRequest(final int size, final Query query) {
+    final var sourceFilter =
+        new SourceFilter.Builder()
+            .includes(
+                PostImporterQueueTemplate.KEY,
+                PostImporterQueueTemplate.POSITION,
+                PostImporterQueueTemplate.INTENT)
+            .build();
+
+    return new SearchRequest.Builder()
+        .index(pendingUpdateAlias)
```

Comment:
:question: is this the same as before?

Reply:
Yes, you were supposed to start here, but I wasn't very clear :smile:

Reply:
yes I realized it after reviewing this PR  :laughing:
I was referring if this code is new or is taken from a current implementation